### ISSUE TITLE
Declare `org-roam-capture-templates` with `defcustom`

### ISSUE THREAD

**Issue by myshevchuk** [org-roam/org-roam#922]
### Description

See discussion in the [Slack thread](https://app.slack.com/client/TTPSWL02H/CV160S8EL/thread/CV160S8EL-1594245213.449000)

@jethrokuan, is `plain`still the only entry type and `org-roam-capture--get-point` the only destination to use in `org-capture-templates`?

**Comment by myshevchuk** [comment]
Also, are `:file-name` and `:header` and `:unnarrowed t` strictly required for `org-roam-caputre` to function properly?

These elements can be enforced in Customize, and such an enforcement will eliminate potential sources of confusion for those using the interface, while requiring zero changes to `org-roam-capture` internal logic.

**Comment by myshevchuk** [comment]
This is an example of how the Customize buffer would look like for the default template
``` el
(("d" "default" plain (function org-roam-capture--get-point)
     "%?"
     :file-name "%<%Y%m%d%H%M%S>-${slug}"
     :head "#+title: ${title}\n"
     :unnarrowed t))
```
Note that `plain`, `(function org-roam-capture--get-point)` and `:unnarrowed t` are completely hidden from the user, while `:file-name` and `:header` are enforced. 

<img width="577" alt="Screenshot 2020-07-11 at 14 26 58" src="https://user-images.githubusercontent.com/24212814/87224075-97e8c280-c382-11ea-9dac-09cf73eaf727.png">

**Comment by myshevchuk** [comment]
Or like this, which would look slightly better for template/header strings containing newlines.

<img width="637" alt="Screenshot 2020-07-11 at 14 50 00" src="https://user-images.githubusercontent.com/24212814/87224479-f82d3380-c385-11ea-878e-5c51ce730d74.png">

**Comment by zaeph** [comment]
Closed by #924, and similar issues tracked in #895.
